ml/src/test/scala/org/apache/spark/ml/spark/models/H2OSupervisedMOJOModelTestSuite.scala

```diff
@@ -0,0 +1,168 @@
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.apache.spark.ml.spark.models
+
+import ai.h2o.sparkling.ml.algos.{H2ODeepLearning, H2OGBM, H2OGLM, H2OSupervisedAlgorithm}
+import ai.h2o.sparkling.ml.models.H2OSupervisedMOJOModel
+import ai.h2o.sparkling.ml.params.NullableStringParam
+import hex.Model
+import org.apache.spark.SparkContext
+import org.apache.spark.h2o.{H2OBaseModel, H2OBaseModelBuilder}
+import org.apache.spark.h2o.utils.{SharedH2OTestContext, TestFrameUtils}
+import org.apache.spark.ml.{Pipeline, PipelineModel}
+import org.apache.spark.sql.Row
+import org.junit.runner.RunWith
+import org.scalatest.{FunSuite, Matchers}
+import org.scalatest.junit.JUnitRunner
+import water.api.TestUtils
+
+@RunWith(classOf[JUnitRunner])
+class H2OSupervisedMOJOModelTestSuite extends FunSuite with Matchers with SharedH2OTestContext {
+
+  override def createSparkContext = new SparkContext("local[*]", getClass.getSimpleName, conf = defaultSparkConf)
+
+  private lazy val dataset = spark.read
+    .option("header", "true")
+    .option("inferSchema", "true")
+    .csv(TestUtils.locate("smalldata/prostate/prostate.csv"))
+    .cache()
+
+  private lazy val Array(trainingDataset, testingDataset) = dataset.randomSplit(Array(0.8, 0.2), 1234L).map(_.cache())
+
+  def testOffsetColumnGetsPropagatedToMOJOModel(
+      algo: H2OSupervisedAlgorithm[_ <: H2OBaseModelBuilder, _ <: H2OBaseModel, _ <: Model.Parameters]): Unit = {
+    val offsetColumn = "PSA"
+    algo
+      .setSplitRatio(0.8)
+      .setSeed(1)
+      .setFeaturesCols("AGE", "RACE", "DPROS", "DCAPS", "PSA", "VOL", "GLEASON")
+      .setLabelCol("CAPSULE")
+      .setOffsetCol(offsetColumn)
+
+    val pipeline = new Pipeline().setStages(Array(algo))
+
+    val model = pipeline.fit(dataset)
+    val path = s"ml/build/offset_propagation_to_mojo_${algo.getClass.getSimpleName}"
+    model.write.overwrite().save(path)
+    val loadedModel = PipelineModel.load(path)
+
+    def getModelOffset(model: PipelineModel): String = {
+      model.stages(0).asInstanceOf[H2OSupervisedMOJOModel].getOffsetCol()
+    }
+    val modelOffset = getModelOffset(model)
+    val deserializedModelOffset = getModelOffset(loadedModel)
+
+    modelOffset shouldEqual offsetColumn
+    deserializedModelOffset shouldEqual offsetColumn
+  }
+
+  test("Offset column gets propagated to MOJO model - GBM") {
+    testOffsetColumnGetsPropagatedToMOJOModel(new H2OGBM())
+  }
+
+  test("Offset column gets propagated to MOJO model - GLM") {
+    testOffsetColumnGetsPropagatedToMOJOModel(new H2OGLM())
+  }
+
+  test("Offset column gets propagated to MOJO model - DeepLearning") {
+    testOffsetColumnGetsPropagatedToMOJOModel(new H2ODeepLearning())
```

Comment:
Is Model offset available also on XGBoost?

Reply:
It needs to be implemented on the H2O-3 side first. https://0xdata.atlassian.net/browse/PUBDEV-7029